src/infra/core/src/use_cases/create_dataset_from_snapshot_use_case_impl.rs

```diff
@@ -71,12 +77,7 @@ impl CreateDatasetFromSnapshotUseCase for CreateDatasetFromSnapshotUseCaseImpl {
                 MESSAGE_PRODUCER_KAMU_CORE_DATASET_SERVICE,
                 DatasetLifecycleMessage::created(
                     create_dataset_result.dataset_handle.id.clone(),
-                    match self.current_account_subject.as_ref() {
-                        CurrentAccountSubject::Anonymous(_) => {
-                            panic!("Anonymous account cannot create dataset");
-                        }
-                        CurrentAccountSubject::Logged(l) => l.account_id.clone(),
-                    },
+                    logged_account_id,
                     options.dataset_visibility,
```

Comment:
I'm not sure about passing visibility in a message here... it's like we're using a message to pass initial ReBAC properties to another layer.

I guess we consider ReBAC in a different domain from datasets, so we can't both create a dataset and set properties in a single DB transaction, is that the reasoning here?

Reply:
Will be covered by #980

Reply:
We discussed this point with @zaychenko-sergei  : at the moment `dataset_visibility` is sent this way, but further on, during the big split, @zaychenko-sergei will remove this.

Thus, for the time being, it is proposed to leave it as it is.